### PR TITLE
Support of S/MIME for Google CSE GMail

### LINKED ISSUE #109: Support for GMail S/MIME requirements
Google CSE provides encrypted mail support through Workspace CSE.

Certificates used must obey strict format rules described in [this Google document](https://support.google.com/a/answer/7300887?fl=1&sjid=2466928410660190479-NA)

This issue requires completing #108 first

### PR DESCRIPTION
**Pull request by ThibsG**
- [x] Add endpoint CSE /privatekeysign (for sending signed email)
- [x] Add endpoint CSE /privatekeydecrypt (for encrypting email)
- [x] Generate RSA Certificate/Private key via openssl
- [ ] Generate RSA Certificate/Private key via `ckms`
- [ ] Generate keypair and identity for users via `ckms`

closes #109

### CODE REVIEW

File: crate/server/src/routes/google_cse/operations.rs [hunk 1]
```diff
@@ -193,7 +220,261 @@ pub async fn unwrap(
...
+    }
+    let allocation_size = ctx.decrypt(&encrypted_dek, None)?;
+    debug!("privatekeydecrypt: allocation_size: {allocation_size}");
+    let mut plaintext = vec![0_u8; allocation_size];
+    let decrypt_size = ctx.decrypt(&encrypted_dek, Some(&mut *plaintext))?;
+
+    kms_ensure!(
+        decrypt_size > 0,
+        "privatekeydecrypt: decrypted plaintext cannot be empty"
+    );
```
Comment (ccorsin) [inline crate/server/src/routes/google_cse/operations.rs:412]: Why do we need this specific check ? We should check the lenght of the encrypted text instead no? If plaintext is empty - it means that encrypted text was empty first

File: crate/server/src/routes/google_cse/operations.rs [hunk 2]
```diff
@@ -193,7 +220,261 @@ pub async fn unwrap(
...
+    /// Base64-encoded message digest.
+    /// The digest of the DER encoded SignedAttributes.
+    /// This value is unpadded. Max size: 128B
+    pub digest: String,
+
+    /// The format of the private key or the wrapped private key is up to
+    /// the Key Access Control List Service (KACLS) implementation.
+    /// On the client and on the Gmail side, this is treated as an opaque blob.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub e_key: Option<String>,
```
Comment (ccorsin) [inline crate/server/src/routes/google_cse/operations.rs:250]: remove this param ?

File: crate/server/src/tests/google_cse/mod.rs [hunk 3]
```diff
@@ -0,0 +1,292 @@
...
+    let wrapped_private_key =
+        include_str!("../../../../../documentation/docs/google_cse/blue_wrapped_private_key");
+    let digest = "MSKaRPMiIFwZoWGYjA/MV8mLNNYGW3GpODrEjdbbQqE=";
+
+    let pksr = PrivateKeySignRequest {
+        authentication: jwt.clone(),
+        authorization: jwt,
+        algorithm: "SHA256withRSA".to_string(),
+        digest: digest.to_string(),
+        e_key: Some("e_key".to_string()),
```
Comment (ccorsin) [inline crate/server/src/tests/google_cse/mod.rs:176]: remove this

Comment (Manuthor) [reply crate/server/src/tests/google_cse/mod.rs]: done in a83e9638f1de6b310f771e776498c958c1d9f7d6

File: crate/server/src/routes/google_cse/operations.rs (hunk 2 above)
Comment (Manuthor) [reply crate/server/src/routes/google_cse/operations.rs]: done in a83e9638f1de6b310f771e776498c958c1d9f7d6

**Pull request merged**